code/frameworks/angular/src/client/public-types.ts

```diff
@@ -21,27 +23,54 @@ export type { AngularRenderer };
  *
  * @see [Default export](https://storybook.js.org/docs/formats/component-story-format/#default-export)
  */
-export type Meta<TArgs = Args> = ComponentAnnotations<AngularRenderer, TransformEventType<TArgs>>;
+export type Meta<TArgs = Args> = ComponentAnnotations<
+  AngularRenderer,
+  TransformComponentType<TArgs>
+>;
 
 /**
  * Story function that represents a CSFv2 component example.
  *
  * @see [Named Story exports](https://storybook.js.org/docs/formats/component-story-format/#named-story-exports)
  */
-export type StoryFn<TArgs = Args> = AnnotatedStoryFn<AngularRenderer, TransformEventType<TArgs>>;
+export type StoryFn<TArgs = Args> = AnnotatedStoryFn<
+  AngularRenderer,
+  TransformComponentType<TArgs>
+>;
 
 /**
  * Story object that represents a CSFv3 component example.
  *
  * @see [Named Story exports](https://storybook.js.org/docs/formats/component-story-format/#named-story-exports)
  */
-export type StoryObj<TArgs = Args> = StoryAnnotations<AngularRenderer, TransformEventType<TArgs>>;
+export type StoryObj<TArgs = Args> = StoryAnnotations<
+  AngularRenderer,
+  TransformComponentType<TArgs>
+>;
 
 export type Decorator<TArgs = StrictArgs> = DecoratorFunction<AngularRenderer, TArgs>;
 export type Loader<TArgs = StrictArgs> = LoaderFunction<AngularRenderer, TArgs>;
 export type StoryContext<TArgs = StrictArgs> = GenericStoryContext<AngularRenderer, TArgs>;
 export type Preview = ProjectAnnotations<AngularRenderer>;
 
+/**
+ * Utility type that transforms InputSignal and EventEmitter types
+ */
+type TransformComponentType<T> = TransformInputSignalType<TransformEventType<T>>
+
+// @ts-ignore Angular < 17.2 doesn't export InputSignal
+type AngularInputSignal<T> = AngularCore.InputSignal<T>
+// @ts-ignore Angular < 17.2 doesn't export InputSignalWithTransform
+type AngularInputSignalWithTransform<T, U> = AngularCore.InputSignalWithTransform<T, U>
+
+type AngularHasSignal = typeof AngularCore extends { input: infer U } ? true : false;
+type InputSignal<T> = AngularHasSignal extends true ? AngularInputSignal<T> : never;
+type InputSignalWithTransform<T, U> = AngularHasSignal extends true ? AngularInputSignalWithTransform<T, U> : never;
+
+type TransformInputSignalType<T> = {
+   [K in keyof T]: T[K] extends InputSignal<infer E> ? E : T[K] extends InputSignalWithTransform<any, infer U> ? U : T[K];
```

Comment:
Should this be 

```suggestion
   [K in keyof T]?: T[K] extends InputSignal<infer E> ? E : T[K] extends InputSignalWithTransform<any, infer U> ? U : T[K];
```

because `input.required()` has no `| undefined` and might cause odd story author type errors? I might be wrong

Reply:
That's not necessary, because the `StoryAnnotations` take in theory a third generic `TArgsRequired`, which is not defined in the Angular types, so that all arguments are per default optional.